docs/proposals/secure-impersonation.md

```diff
@@ -0,0 +1,636 @@
+# Secure Impersonation
+
+* [Context](#context)
+* [Goals](#goals)
+* [Design](#design)
+  - [controller serviceAccounts](#controller-serviceaccounts)
+  - [user/group names](#user-group-names)
+  - [serviceAccount impersonation](#serviceaccount-impersonation)
+  - [defaulting](#defaulting)
+  - [kubeConfig](#kubeconfig)
+  - [sourceRefs](#sourcerefs)
+  - [kubectl](#kubectl)
+* [Concerns](#concerns)
+  - [compatibiliity](#compatibiliity)
+  - [performance](#performance)
+* [Decisions](#decisions)
+* [Tasks](#tasks)
+* [Examples](#examples)
+  - [default install](#default-install)
+  - [tenants](#tenants)
+  - [cross-namespace binding against Flux Groups](#cross-namespace-binding-against-flux-groups)
+  - [opening Source policies with RBAC](#opening-source-policies-with-rbac)
+  - [rejecting kubeConfigs](#rejecting-kubeconfigs)
+
+## Context
+
+Flux 2's installation model is built from CRD's.
+This allows cluster operators to granularly control who can create/modify Sources and Applier type objects such as
+GitRepository and Kustomization and for which Namespaces.
+This also helps out the Deployment model.
+While we could for example, still run independent instances of ex:kustomize-controller in each Namespace listening to only
+Kustomize resources for that Namespace, it can use fewer resources to deploy a single controller watching all (or many) Namespaces.
+Having a single controller reduces the onboarding steps necessary for a new Namespace owner to use Flux Custom Resources.
+A central set of controllers is the primary Deployment mode for Flux.
+
+In order to manage resources across many Namespaces, the default install gives kustomize-controller and helm-controller a full
+cluster-admin ClusterRoleBinding.
+This means cluster owners need to be careful who can create Kustomizations and HelmReleases as well as which Sources are in the cluster
+due to the exposure they provide to act as cluster-admin.
+This restricts who can create Source and Apply type objects into a burdensome approval flow for cluster owners and provides
+poor delegation of self-service permissions which is a proven elements of good platforms.
```

Comment:
This sentence provoked a double-take -- it's not clear from the syntax whether _poor_ delegation of self-service permissions is a proven element of good platforms, or good delegation [...] would be. The meaning is obvious after the fact, but perhaps it could be rewritten to flow better.